src/backend/src/services/change-requests.services.ts

```diff
@@ -614,87 +612,114 @@ export default class ChangeRequestsService {
-        throw new HttpException(
-          400,
-          'Change Request with proposed changes must have either project or work package proposed changes'
-        );
-      } else {
-        const { name, status, projectLeadId, projectManagerId, links } = wbsProposedChanges;
+    if (projectProposedChanges && workPackageProposedChanges) {
+      throw new HttpException(400, "Change Request can't be on both a project and a work package");
+    } else if (!projectProposedChanges && !workPackageProposedChanges) {
+      throw new HttpException(
+        400,
+        'Change Request with proposed changes must have either project or work package proposed changes'
+      );
+    } else if (projectProposedChanges) {
+      const { name, status, projectLeadId, projectManagerId, links } = projectProposedChanges;
+      const { budget, summary, newProject, rules, teamIds, goals, features, otherConstraints } = projectProposedChanges;
 
-        if (projectLeadId) {
-          const projectLead = await prisma.user.findUnique({ where: { userId: projectLeadId } });
-          if (!projectLead) throw new NotFoundException('User', projectLeadId);
+      if (projectLeadId) {
+        const projectLead = await prisma.user.findUnique({ where: { userId: projectLeadId } });
+        if (!projectLead) throw new NotFoundException('User', projectLeadId);
+      }
+
+      if (projectManagerId) {
+        const projectManager = await prisma.user.findUnique({ where: { userId: projectManagerId } });
+        if (!projectManager) throw new NotFoundException('User', projectManagerId);
+      }
+
+      if (links.length > 0) {
+        for (const link of links) {
+          const linkType = await prisma.linkType.findUnique({ where: { name: link.linkTypeName } });
+          if (!linkType) throw new NotFoundException('Link Type', link.linkTypeName);
         }
+      }
 
-        if (projectManagerId) {
-          const projectManager = await prisma.user.findUnique({ where: { userId: projectManagerId } });
-          if (!projectManager) throw new NotFoundException('User', projectManagerId);
+      if (teamIds.length > 0) {
+        for (const teamId of teamIds) {
+          const team = await prisma.team.findUnique({ where: { teamId } });
+          if (!team) throw new NotFoundException('Team', teamId);
         }
+      }
 
-        if (links.length > 0) {
-          for (const link of links) {
-            const linkType = await prisma.linkType.findUnique({ where: { name: link.linkTypeName } });
-            if (!linkType) throw new NotFoundException('Link Type', link.linkTypeName);
+      const createdProposedChanges = await prisma.wbs_Proposed_Changes.create({
+        data: {
+          changeRequestId: createdCR.scopeChangeRequest!.scopeCrId,
+          name,
+          status,
+          projectLeadId,
+          projectManagerId,
+          links: {
+            create: links.map((linkInfo) => ({ url: linkInfo.url, linkTypeName: linkInfo.linkTypeName }))
```

Comment:
you should be able to do a similar create for the projectproposedsolutions and do everything for this if statement in one prisma call (same idea for workpackageproposedsolutions)